app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ListProspects.php

```diff
@@ -151,6 +154,25 @@ function (Builder $query) {
                                 ->get();
                         }
                     ),
+                SelectFilter::make('tags')
+                    ->label('Tags')
+                    ->options(fn (): array => Tag::query()->where('type', TagType::Prospect)->pluck('name', 'id')->toArray())
+                    ->searchable()
+                    ->preload()
+                    ->optionsLimit(20)
+                    ->multiple()
+                    ->query(
+                        function (Builder $query, array $data) {
+                            if (blank($data['values'])) {
+                                return;
+                            }
+
+                            $query->whereHas('tags', function (Builder $query) use ($data) {
+                                $query->whereIn('tag_id', $data['values'])->where('taggable_type', 'prospect');
```

Comment:
This should be checking the tag `id` and tag `type` rather than the pivot I think.